proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+  }
+
+
+  //////////////// Lexical productions listed first, support functions follow.
+  public ProToken nextToken() {
+    LOGGER.trace("Entering nextToken()");
+    try {
+    for (;;) {
+
+      if (preserve) {
+        // The preserved character is the character prior to currChar.
+        textStartFile = preserveFile;
+        textStartLine = preserveLine;
+        textStartCol = preserveCol;
+        textStartSource = preserveSource;
+        currText.setLength(1);
+        currText.setCharAt(0, (char) preserveChar);
+        preserveDrop(); // we are done with the preservation
+        switch (preserveChar) {
+          case '.':
+            return periodStart();
+          case ':':
+            return colon();
+        } // switch
+      }
+
+      // Proparse Directive
+      // Check this before setting currText...
+      // we don't want BEGIN_PROPARSE_DIRECTIVE in the text
+      if (currInt == ProgressLexer.PROPARSE_DIRECTIVE) {
+        textStartFile = prepro.getTextStart().getFile();
+        textStartLine = prepro.getTextStart().getLine();
+        textStartCol = prepro.getTextStart().getCol();
+        textStartSource = prepro.getTextStart().getSourceNum();
+        getChar();
+        return makeToken(PreprocessorParser.PROPARSEDIRECTIVE, prepro.getProparseDirectiveText());
+      }
+
+      textStartFile = prepro.getFileIndex();
+      textStartLine = prepro.getLine2();
+      textStartCol = prepro.getColumn();
+      textStartSource = prepro.getSourceNum();
+      currText.setLength(1);
+      currText.setCharAt(0, (char) currInt);
+
+      if (gettingAmpIfDefArg) {
+        getChar();
+        gettingAmpIfDefArg = false;
+        return ampIfDefArg();
+      }
+
+      switch (currChar) {
+
+        case '\t':
+        case '\n':
+        case '\f':
+        case '\r':
+        case ' ':
+          getChar();
+          return whitespace();
+
+        case '"':
+        case '\'':
+          if (prepro.isEscapeCurrent()) {
+            getChar();
+            // Escaped quote does not start a string
+            return id(PreprocessorParser.FILENAME);
+          } else {
+            currStringType = currInt;
+            getChar();
+            return quotedString();
+          }
+
+        case '/':
+          getChar();
+          if (currChar == '*') {
+            return comment();
+          } else if (currChar == '/') {
+            return singleLineComment();
+          } else if (currChar == '(' || currIsSpace()) {
+            // slash (division) can only be followed by whitespace or '('
+            // ...that's what I found empirically, anyway. (jag 2003/05/09)
+            return makeToken(PreprocessorParser.SLASH);
+          } else {
+            append();
+            getChar();
+            return id(PreprocessorParser.FILENAME);
+          }
+
+        case ':':
+          getChar();
+          return colon();
+
+        case '&':
+          getChar();
+          return ampText();
+        case '@':
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1151)